auth/HOSTED.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+#
+# Copyright 2021 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# Configure an environment to run Apigee Registry clients with a Google-hosted service.
+#
+# The following assumes you have run `gcloud auth login` and that the current
+# gcloud project is the one with your Apigee Registry instance.
+#
+
+if ! [ -x "$(command -v gcloud)" ]; then
+  echo 'ERROR: This script requires the gcloud command. Please install it to continue.' >&2; return
+fi
+
+# Calls to the hosted service are secure.
+unset APG_REGISTRY_INSECURE
+
+# Get the service address.
+export APG_REGISTRY_AUDIENCES=https://apigeeregistry.googleapis.com
+export APG_REGISTRY_ADDRESS=${APG_REGISTRY_AUDIENCES#https://}:443
+
+# The auth token is generated for the gcloud logged-in user.
+export APG_REGISTRY_CLIENT_EMAIL=$(gcloud config list account --format "value(core.account)")
```

Comment:
Consider `ACCOUNT` instead of `EMAIL` if this is going to be used as a config value. That seems to be the terminology used by gcloud, although I think accounts might always be formatted as email addresses.

If it's just a variable meant to be used on the following line, let's keep it unexported.

Reply:
Worth discussing; I've been looking for the original source that I used for this and haven't found anything using "EMAIL" in the variable name. That is consistently used in multiple files in the `auth` directory, so I prefer not to change it here, but wouldn't rule it out.